dash/development/_r_components_generation.py

```diff
@@ -625,3 +641,104 @@ def generate_exports(
         package_imports,
         package_suggests,
     )
+
+
+def get_r_prop_types(type_object):
+    """Mapping from the PropTypes js type object to the R type"""
+
+    def shape_or_exact():
+        return 'lists containing elements {}.\n{}'.format(
+            ', '.join(
+                "'{}'".format(t) for t in list(type_object['value'].keys())
+            ),
+            'Those elements have the following types:\n{}'.format(
+                '\n'.join(
+                    create_prop_docstring(
```

Comment:
oh hmm - `create_prop_docstring` means nested items are going to get python type info rather than R, right? It might be possible to pass an optional type conversion function into `create_prop_docstring`, if `get_r_type ` behaves compatibly with `js_to_py_type` (which it looks like it does...)

https://github.com/plotly/dash/blob/4deb9000f3c056ba1022772fb5479b3d4f647048/dash/development/_py_components_generation.py#L421

Reply:
fixed in a68ace3342c7fdb2fa36e217bb6162670bb130f1

Reply:
That'll work. I suspect we could do it 🌴 using the same function just passing in another arg to replace `js_to_py_type` - but we can come back to that after we have tests.